src/main/java/wolox/reactortraining/services/BotService.java

```diff
@@ -38,4 +99,56 @@
             .bodyToMono(BotResponse.class);
     }
 
+    @Override
+    public Mono<Void> createBot(BotCreationDto botCreationDto) {
+        Mono<User> userMono = createUser(botCreationDto.getUsername());
+        Mono<Void> createBot = createBot(botCreationDto.getBotName(), botCreationDto.getTopics());
+
+        return userMono
+            .flatMap(
+                user -> Mono.zip(createTopics(user, botCreationDto.getTopics()), Mono.just(user)))
+            .flatMap(this::addTopicsToUser)
+            .then(createBot);
+    }
+
+    private Mono<User> createUser(String username) {
+        UserDto userDto = new UserDto();
+        userDto.setUsername(username);
+
+        User user = new User(userDto);
+
+        return userRepository.insert(user);
+    }
+
+    private Mono<List<Topic>> createTopics(User user, List<String> topics) {
+        return Flux
+            .fromIterable(topics)
+            .map(BotService::createTopicDto)
+            .map(dto -> createTopicModel(dto, user))
+            .flatMap(topicRepository::insert)
+            .collectList();
+    }
+
+    private Mono<Void> createBot(String botName, List<String> topics) {
+        return twitterService
+            .getTweetsStreamPipe()
+            .map(Tweet::getText)
+            .take(500)
```

Comment:
put 500 value in constant or environment var, how you prefer

Reply:
I gonna add it to `tweeter.properties`